web/src/components/apps/LicenseFields.tsx

```diff
@@ -0,0 +1,135 @@
+  entitlementsToShow: string[];
+}) => {
+  return (
+    <CustomerLicenseFields className="flex flexWrap--wrap">
+      {entitlements?.map((entitlement) => {
+        const currEntitlement = entitlementsToShow?.find(
+          (f) => f === entitlement.title
+        );
+        const isTextField = entitlement.valueType === "Text";
+        const isBooleanField = entitlement.valueType === "Boolean";
+        if (
+          entitlement.value.length > 100 &&
+          currEntitlement !== entitlement.title
+        ) {
+          return (
+            <CustomerLicenseField
+              key={entitlement.label}
+              className={`u-fontSize--small u-lineHeight--normal u-textColor--secondary u-fontWeight--medium u-marginRight--10 u-marginLeft--5`}
+            >
+              {" "}
+              {entitlement.title}:{" "}
+              <span
+                className={`u-fontWeight--bold ${
+                  isTextField && "u-fontFamily--monospace"
+                }`}
+              >
+                {" "}
+                {entitlement.value.slice(0, 100) + "..."}{" "}
+              </span>
+              <span
+                className="replicated-link"
+                onClick={() => toggleShowDetails(entitlement.title)}
+              >
+                show
+              </span>
+            </CustomerLicenseField>
+          );
+        } else if (
+          entitlement.value.length > 100 &&
+          currEntitlement === entitlement.title
+        ) {
+          return (
+            <CustomerLicenseField
+              key={entitlement.label}
+              className={`u-fontSize--small u-lineHeight--normal u-textColor--secondary u-fontWeight--medium u-marginRight--10 u-marginLeft--5`}
+            >
+              {" "}
+              {entitlement.title}:{" "}
+              <span
+                className={`u-fontWeight--bold ${
+                  isTextField && "u-fontFamily--monospace"
+                }`}
+              >
+                {" "}
+                {entitlement.value}{" "}
+              </span>
+              <span
+                className="replicated-link"
+                onClick={() => toggleHideDetails(entitlement.title)}
+              >
+                hide
+              </span>
+            </CustomerLicenseField>
+          );
+        } else {
+          return (
+            <CustomerLicenseField
+              key={entitlement.label}
+              className={`u-fontSize--small u-lineHeight--normal u-textColor--secondary u-fontWeight--medium u-marginRight--10 u-marginLeft--5`}
+            >
+              {" "}
+              {entitlement.title}:{" "}
+              <span
+                className={`u-fontWeight--bold ${
+                  isTextField && "u-fontFamily--monospace"
+                }`}
+              >
+                {" "}
+                {isBooleanField
+                  ? entitlement.value.toString()
+                  : entitlement.value}{" "}
```

Comment:
```suggestion
                {entitlement.value.toString() + " "}
```
could do something like this. strings have `toString()` on them.